src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -231,7 +232,7 @@ private static void HtmlDecode(string value, StringBuilder output)
                         }
                         else
                         {
-                            string entity = value.Substring(entityOffset, entityLength);
+                            ReadOnlySpan<char> entity = valueSpan.Slice(entityOffset, entityLength);
```

Comment:
Rather than storing `value.AsSpan()` into a local and then using `valueSpan.Slice(offset, count)` in these three places, you can just use `value.AsSpan(offset, count)` in these three place.